File: 0651-4-keys-keyboard/0651-4-keys-keyboard.cs
public class Solution {
    public int MaxA(int n) {
        int[] dp = new int[n+1];
        for(int i = 1; i <= n; ++i) {
            dp[i] = i;
            for(int j = 1; j < i-2; ++j) {
                int total = dp[j] + dp[j]*(i-j-2);
                dp[i] = Math.Max(dp[i], total);
            }
        }
        return dp[n];
    }
}

File: 0651-4-keys-keyboard/0651-4-keys-keyboard.py
class Solution:
    def maxA(self, n: int) -> int:
        dp = [0] * (n+1)
        for i in range(1,n+1):
            dp[i] = i
            for j in range(1, i-2):
                total = dp[j] + dp[j] * (i-j-2)
                dp[i] = max(dp[i], total)
        return dp[n]